datafusion/core/src/execution/session_state.rs

```diff
@@ -976,6 +872,713 @@ impl SessionState {
     }
 }
 
+/// A builder to be used for building [`SessionState`]'s. Defaults will
+/// be used for all values unless explicitly provided.
```

Comment:
👍 

We could also add a doc comment to `SessionState` for an example. Something like

```suggestion
/// be used for all values unless explicitly provided.
///
/// See example on [`SessionState`]
```